timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -254,7 +286,93 @@ def _fit(
+                optim="adamw_torch_fused",
+                logging_dir=str(output_dir),
+                logging_strategy="steps",
+                logging_steps=self.logging_steps,
+                save_strategy="steps",
+                save_steps=self.eval_and_save_steps,
+                evaluation_strategy="steps",
+                eval_steps=self.eval_and_save_steps,
+                report_to="none",
+                max_steps=self.fine_tune_steps,
+                gradient_accumulation_steps=1,
+                dataloader_num_workers=self.data_loader_num_workers,
+                tf32=self._has_tf32(),
+                save_only_model=True,
+                load_best_model_at_end=True,
+                metric_for_best_model="eval_loss",
+                prediction_loss_only=True,
+                save_total_limit=1,
+            )
+            tokenizer_train_dataset = ChronosFineTuningDataset(
+                tokenizer=self.model_pipeline.tokenizer,
+                target_df=train_data,
+                freq=self.freq,
+                target_column=self.target,
+                context_length=context_length,
+                prediction_length=self.prediction_length,
+                mode="training",
+            )
+            callbacks = [TimeLimitCallback(time_limit=fine_tune_time_limit), EvaluateAndSaveFinalStepCallback()]
+
+            tokenizer_val_dataset = (
+                ChronosFineTuningDataset(
+                    tokenizer=self.model_pipeline.tokenizer,
+                    target_df=val_data,
+                    freq=self.freq,
+                    target_column=self.target,
+                    context_length=context_length,
+                    prediction_length=self.prediction_length,
+                    mode="validation",
+                )
+                if val_data is not None
+                else None
+            )
+
+            trainer = Trainer(
+                model=self.model_pipeline.model.model,
+                args=training_args,
+                train_dataset=tokenizer_train_dataset,
+                eval_dataset=tokenizer_val_dataset,
+                callbacks=callbacks,
+            )
+
+            if val_data is not None:
+                # Evaluate once before training
+                zero_shot_eval_loss = trainer.evaluate()["eval_loss"]
+
+            trainer.train()
+
+            log_history_df = pd.DataFrame(trainer.state.log_history)
+            best_train_eval_loss = log_history_df["eval_loss"].min()
+
+            if best_train_eval_loss <= zero_shot_eval_loss:
+                fine_tuned_ckpt_path = Path(self.path) / self.fine_tuned_ckpt_name
+                logger.debug(
+                    f"Validation loss improved after fine-tuning. Saving fine-tuned model to {fine_tuned_ckpt_path}"
+                )
+                self.model_pipeline.model.model.save_pretrained(Path(self.path) / self.fine_tuned_ckpt_name)
+
+            if not self.keep_transformers_logs:
```

Comment:
I might have missed it, but I think the `model_path` needs to be overwritten somewhere here so that the Predictor will pick up the fine tuned checkpoint during inference?

Reply:
`load` checks if there's a fine-tuned checkpoint and loads that one. I just realized that reversion to pretrained ckpt will not work if the object is in memory. Fixed that.